asm/f64/stubs_test.go

```diff
@@ -111,16 +120,118 @@ func TestCumProd(t *testing.T) {
 	} {
 		ret := CumProd(v.dst, v.src)
 		for i := range v.expect {
-			if ret[i] != v.expect[i] && (math.IsNaN(ret[i]) != math.IsNaN(v.expect[i])) {
+			if diff(ret[i], v.expect[i]) {
 				t.Log("Test", j, "CumProd error at", i, "Got:", ret[i], "Expected:", v.expect[i])
 				t.Fail()
 			}
-			if ret[i] != v.dst[i] && (math.IsNaN(ret[i]) != math.IsNaN(v.dst[i])) {
+			if diff(ret[i], v.dst[i]) {
 				t.Log("Test", j, "CumProd ret/dst mismatch", i, "Ret:", ret[i], "Dst:", v.dst[i])
 				t.Fail()
 			}
 		}
 	}
+	runtime.GC()
+}
+
+func TestDiv(t *testing.T) {
+	for j, v := range []struct {
+		dst, src, expect []float64
+	}{
+		{[]float64{1}, []float64{1}, []float64{1}},
+		{[]float64{nan}, []float64{nan}, []float64{nan}},
+		{[]float64{1, 2, 3, 4}, []float64{1, 2, 3, 4}, []float64{1, 1, 1, 1}},
+		{[]float64{2, 4, 6}, []float64{1, 2, 3, 4}, []float64{2, 2, 2}},
+		{[]float64{0, 0, 0, 0}, []float64{1, 2, 3}, []float64{0, 0, 0}},
+		{[]float64{nan, 1, nan, 1, 0},
+			[]float64{1, 1, nan, 1, 1},
+			[]float64{nan, 1, nan, 1, 0}},
+		{[]float64{inf, 4, nan, ninf, 9},
+			[]float64{inf, 4, nan, ninf, 3},
+			[]float64{nan, 1, nan, nan, 3}},
+	} {
+		Div(v.dst, v.src)
```

Comment:
The `runtime.GC()` calls perform the safety function for out-of-range memory accesses.  It will panic in that test if any memory violations occur, even if the data is simply read and written back without changes, flagging the asm routine that's in violation.